tests/state/test_state_transition.py

```diff
@@ -120,7 +120,7 @@ def test_compute_cycle_transitions(genesis_crystallized_state,
     parent_active_state = genesis_active_state
     parent_block = genesis_block
     block = copy.deepcopy(genesis_block)
-    block.slot_number = 258
+    block.slot_number = config['cycle_length']
```

Comment:
The purpose of this test is to trigger multiple cycle transitions. We should keep as 258
We should also maybe add a comment to the test to say what the intention is here. I used plural "transitions" in the test name but that is not obvious enough

Reply:
Sorry I didn't notice that. I tried to keep `total_deposits` positive. It can be `config['cycle_length'] * 2` (128), but can't be `config['cycle_length'] * 3`. Do you think 128 is enough?

Reply:
hm, I'd like to be able to test passing up a bunch of cycles at once. I'm surprised it goes negative that quick. Also aren't we ensuring that individual deposits don't go below 0?

Reply:
The minimum balance is 0, but if `total_deposits` is 0, then `reward_quotient = config['base_reward_quotient'] * int(sqrt(total_deposits_in_ETH))` is also 0, then we still have a division by zero iproblem when calculating `validators[index].balance // reward_quotient`. I'm okay with setting a temporary higher threshold that is greater than 0 with clear comment. (Maybe 1?)
```python
        # TODO: Now we temporarily force the balance positive until we have clear rule of forced exit.
        if updated_validators[index].balance < 0:
            updated_validators[index].balance = 1
```

Reply:
I'm mainly concerned that validator deposits are going to zero after just being offline for a few cycles. We can do the temp setting to 1, but we'll need to investigate why the balances are dropping so quickly. One of the constants might be off

Reply:
I'm going to pull this down and investigate

Reply:
Quadratic leaking strongly affects penalty. 

The balance `B` is `3.2E+19`, `reward_quotient` is `1146880`.


1. When `slot = config['cycle_length'] * 2`, then `time_since_finality <= 2 * config['cycle_length']`:

Slot | rewards_and_penalties | cumulative rewards_and_penalties | B + cumulative   rewards_and_penalties
-- | -- | -- | --
0 | -2.79018E+13 | -2.79018E+13 | 3.2E+19
1 | -5.58036E+13 | -8.37054E+13 | 3.19999E+19
2 | -8.37054E+13 | -1.67411E+14 | 3.19998E+19
3 | -1.11607E+14 | -2.79018E+14 | 3.19997E+19
4 | -1.39509E+14 | -4.18527E+14 | 3.19996E+19


2. When `slot = config['cycle_length'] * 3` then `time_since_finality > 2 * config['cycle_length']`:
    - `time_since_finality`: `192`
    - `quadratic_penalty_quotient`: `362`

Slot | rewards_and_penalties | cumulative rewards_and_penalties | B + cumulative   rewards_and_penalties
-- | -- | -- | --
0 | -1.69724E+19 | -1.69724E+19 | 1.50276E+19
1 | -3.39448E+19 | -5.09172E+19 | -1.89172E+19
2 | -5.09172E+19 | -1.01834E+20 | -6.98344E+19
3 | -6.78896E+19 | -1.69724E+20 | -1.37724E+20
4 | -8.4862E+19 | -2.54586E+20 | -2.22586E+20

Our test case is extream because everyone was offline for a long time. Does the numbers make sense to you?